mujoco_ros/src/offscreen_camera.cpp

```diff
@@ -298,15 +301,15 @@ void OffscreenCamera::renderAndPublish(mujoco_ros::OffscreenRenderContext *offsc
 		rendered                            = renderAndPubIfNecessary(offscreen, segment, true, segment);
 	}
 
-	if (rendered) {
-		publishCameraInfo();
+	if (!rendered) {
+		publishCameraInfo(ros::Time::now());
 	}
 }
 
-void OffscreenCamera::publishCameraInfo()
+void OffscreenCamera::publishCameraInfo(const ros::Time &t)
 {
 	sensor_msgs::CameraInfo camera_info_msg = camera_info_manager_->getCameraInfo();
-	camera_info_msg.header.stamp            = ros::Time::now();
```

Comment:
Technically this shouldn't make a difference. The physics loop waits until no offscreen render request are pending, thus ros::Time(scn_state_.data.time) should be equal to ros::Time::now()

Reply:
I think you assumption does not hold:

Without the patch i get this:

```
rostopic echo /mujoco_server/cameras/camera/rgb | grep -B 1 nsecs:
--
    secs: 109
    nsecs: 276000000
--
    secs: 109
    nsecs: 344000000
--
    secs: 109
    nsecs: 480000000
--
    secs: 109
    nsecs: 684000000
```

```
rostopic echo /mujoco_server/cameras/camera/camera_info | grep -B 1 nsecs
--
    secs: 109
    nsecs: 210000000
--
    secs: 109
    nsecs: 278000000
--
    secs: 109
    nsecs: 346000000
--
    secs: 109
    nsecs: 414000000
--
    secs: 109
    nsecs: 482000000
--
    secs: 109
    nsecs: 550000000
--
    secs: 109
    nsecs: 618000000
--
    secs: 109
    nsecs: 686000000
--
    secs: 109
    nsecs: 754000000
--
    secs: 109
    nsecs: 822000000
--
    secs: 109
    nsecs: 890000000
--
    secs: 109
    nsecs: 958000000

```